src/components/general/drag-and-drop/DragAndDrop.jsx

```diff
@@ -0,0 +1,395 @@
+                const quaternion = camera.quaternion.clone(); */
+
+                const items = Array.from( event.dataTransfer.items );
+
+                await Promise.all( items.map( async item => {
+
+                    const drop = _isJsonItem( item );
+                    const app = await uploadCreateApp( item, { drop });
+
+                    if ( app ) {
+
+                        if ( drop ) {
+
+                            world.appManager.importApp( app );
+                            setState({ openedPanel: null });
+
+                        } else {
+
+                            setQueue( queue.concat([ app ]) );
+
+                        }
+
+                    }
+
+                }));
+
+                /* let arrowLoader = metaverseUi.makeArrowLoader();
+                arrowLoader.position.copy(position);
+                arrowLoader.quaternion.copy(quaternion);
+                scene.add(arrowLoader);
+                arrowLoader.updateMatrixWorld();
+
+                if (arrowLoader) {
+                    scene.remove(arrowLoader);
+                    arrowLoader.destroy();
+                } */
+
+            }
+
+        };
+
+        window.addEventListener( 'drop', drop );
+
+        //
+
+        return () => {
+
+            window.removeEventListener( 'dragover', dragover );
+            window.removeEventListener( 'drop', drop );
+
+        };
+
+    }, [] );
+
+    useEffect( () => {
+
+        if ( queue.length > 0 && ! currentApp ) {
+
+            const app = queue[0];
+            setCurrentApp( app );
+            setQueue( queue.slice(1) );
+            setState({ openedPanel: null });
+
+            if ( cameraManager.pointerLockElement ) {
+
+                cameraManager.exitPointerLock();
+
+            }
+
+        }
+
+    }, [ queue, currentApp ] );
+
+    const _currentAppClick = ( event ) => {
+
+        event.preventDefault();
+        event.stopPropagation();
+
+    };
+
+    const _importApp = ( app ) => {
+
+        const localPlayer = metaversefile.useLocalPlayer();
+        const position = localPlayer.position.clone().add( new THREE.Vector3( 0, 0, - 2 ).applyQuaternion( localPlayer.quaternion ) );
+        const quaternion = localPlayer.quaternion;
+
+        app.position.copy( position );
+        app.quaternion.copy( quaternion );
+        app.updateMatrixWorld();
+        world.appManager.importApp( app );
+
+    };
+
+    const _drop = async ( event ) => {
+
+        event.preventDefault();
+        event.stopPropagation();
+
+        if ( currentApp ) {
+
+            _importApp( currentApp );
+            setCurrentApp( null );
+
+        }
+
+    };
+
+    const _equip = ( event ) => {
+
+        event.preventDefault();
+        event.stopPropagation();
+
+        if ( currentApp ) {
+
+            const app = currentApp;
+            _importApp( app );
+            app.activate();
+            setCurrentApp( null );
+
+        }
+
+    };
+
+    const _mint = ( event ) => {
```

Comment:
this currently does nothing?  if it's going to be updated later, can we add a comment here stating that?

Reply:
Not sure about part of dead code,
I also saw some parts, but don't have enough info about this and only the 'author' of the code knows if the code is not important anymore.

Reply:
ok, we'll leave it for now and find out from the author what the situation is